create-serverless-service/test/test.js

```diff
@@ -15,26 +15,249 @@
+    });
+
+    /*
+    * Given an event with an invalid body.service_name, handler() should indicate service name has specified issues
+    * @param {object} event, contains either a missing or invalid body.service_name property
+    * @params {object, function} default aws context and callback function as assigned above respectively
+    * @returns index.handler() should return a descriptive InternalServerError notification
+    */
+    it("should inform user of error if given an event with an invalid body.service_name", function(){
+      //no characters
+      var invalidName1 = "";
+      //contains a non-alphanumeric character
+      var invalidName2 = "Rar!ty";
+      var nameValues = [null, undefined, invalidName1, invalidName2];
+      var errMessage = "Service name is not defined or has invalid characters";
+      var errType = "InternalServerError";
+      var allCases = true;
+      //if checkCase() returns false for any of the nameValues assigned above, have allCases be false
+      for (i in nameValues){
+        if(!checkCase("body","service_name",nameValues[i],errMessage,errType)){
+          allCases = false;
+        }
+      }
+      assert.isTrue(allCases);
+    });
+
+    /*
+    * Given an event with missing headers/headers.Authorization, handler() should throw InternalServerError
+    * @param {object} event, contains a null or undefined headers or headers.Authorization
+    * @params {object, function} default aws context and callback function as assigned above respectively
+    * @returns index.handler() should return an InternalServerError notification
+    */
+    it("should throw an InternalServerError error if missing event.headers", function(){
+      var errMessage = "Authorization header is missing";
+      var errType = "InternalServerError";
+      var allCases = checkCase("headers", "Authorization", null, errMessage, errType) &&
+                      checkCase("headers", "Authorization", undefined, errMessage, errType) &&
+                      checkCase("headers", null, null, errMessage, errType) &&
+                      checkCase("headers", null, undefined, errMessage, errType);
+      assert.isTrue(allCases);
+    });
 
-describe('Sample', function() {
-    it('tests handler', function(done) {
+    /*
+    * Given an event indicating a lambda or api service but no runtime, handler() informs of missing Runtime
+    * @param {object} event, contains a service type that isn't "website", and no body.runtime
+    * @params {object, function} default aws context and callback function as assigned above respectively
+    * @returns index.handler() should return an InternalServerError notification
+    */
+    it("should inform of error if given no event.body.runtime for a service other than website", ()=>{
+      var runtime = "";
+      var errType = "InternalServerError";
+      var errMessage = "Service Runtime not defined";
+      var allCases = checkCase("body","runtime",runtime,errMessage,errType) &&
```

Comment:
This test is failing on latest code: Can you check?

Reply:
Just fixed, the string that represents the expected error message in the test was out-dated and I gave it the current expected error message.